goredis/broker.go

```diff
@@ -0,0 +1,89 @@
+// Package goredis implements a Celery broker using GoRedis.
+package goredis
+
+import (
+	"context"
+	"github.com/go-redis/redis/v9"
+	"github.com/marselester/gopher-celery/internal/brokertools"
+	"time"
+)
+
+// DefaultReceiveTimeout defines how many seconds the broker's Receive command
+// should block waiting for results from Redis.
+// Larger the timeout, longer the client will have to wait for Celery app to exit.
+const DefaultReceiveTimeout = 5
+
+// BrokerOption sets up a Broker.
+type BrokerOption func(*Broker)
+
+// WithBrokerClient sets Redis connection pool.
+func WithBrokerClient(pool *redis.Client) BrokerOption {
+	return func(c *Broker) {
+		c.pool = pool
+	}
+}
+
+// NewBroker creates a broker backed by Redis.
+// By default, it connects to localhost.
+func NewBroker(options ...BrokerOption) *Broker {
+	br := Broker{
+		receiveTimeout: DefaultReceiveTimeout,
```

Comment:
It should be `DefaultReceiveTimeout * time.Second`.

Reply:
good catch, updated.